lib/zhr_devs/submissions/events/task_downloaded.ex

```diff
@@ -3,11 +3,11 @@ defmodule ZhrDevs.Submissions.Events.TaskDownloaded do
   Represents an event that is emitted when user is allowed to download a task before the first submission.
   """
   alias Uptight.Base.Urlsafe
-  alias Uptight.Text, as: T
+  alias Uptight.Text.Urlencoded
 
   @fields [
     technology: nil,
-    task_id: T.new(),
+    task_id: Urlencoded.new(),
     hashed_identity: Urlsafe.new()
```

Comment:
Now this *is* Urlsafe because the output of hash is raw, unencoded binary!

If you have something that's already encoded as human-readable text (for example hash digest or UUID) you should use `Text` to store it and only drop to `Urlencoded` at the plug level when you send the data down the wire!

Reply:
(This logic of translating data structures while preserving their inner type at the latest possible moment actually applies to all the data structures and all data types! The only reason why we convert raw binaries to the Urlsafe Base64 form is that Erlang doesn't give us an easy way to coerce data types like Lean or Haskell do, so we'd better carry **encoding** together with **raw data** around in the whole app. But suppose you had an interface that only accepts a certain kind of raw binaries, then we wouldn't carry around the transformed data for this adapter, we'd still carry around just the raw data and adapt it at the last possible moment! Remember, we also can have multiple interfaces which require different encodings too!).

Reply:
This deserves a separate pull request, I believe